core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -938,7 +938,7 @@ object KafkaConfig {
 
       /** ********* Replication configuration ***********/
       .define(ControllerSocketTimeoutMsProp, INT, Defaults.ControllerSocketTimeoutMs, MEDIUM, ControllerSocketTimeoutMsDoc)
-      .define(DefaultReplicationFactorProp, INT, Defaults.DefaultReplicationFactor, MEDIUM, DefaultReplicationFactorDoc)
+      .define(DefaultReplicationFactorProp, SHORT, Defaults.DefaultReplicationFactor, MEDIUM, DefaultReplicationFactorDoc)
```

Comment:
This is an incompatible change which would require a KIP and a new major version.